exchanges/kucoin.go

```diff
@@ -0,0 +1,413 @@
+// Copyright © 2017 Alessandro Sanino <saninoale@gmail.com>
+// Copyright © 2018 Mangrovia Solutions
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package exchanges
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dgrr/kucoin-go"
+	"github.com/dgrr/kucoin-go/websocket"
+	"github.com/saniales/golang-crypto-trading-bot/environment"
+	"github.com/shopspring/decimal"
+)
+
+// KucoinWrapper wrapsKucoin
+type KucoinWrapper struct {
+	api         *kucoin.Kucoin
+	ws          *kucoin.WSClient
+	websocketOn bool
+	summaries   *SummaryCache
+	orderbook   *OrderbookCache
+}
+
+// NewKucoinWrapper creates a generic wrapper of theKucoin
+func NewKucoinWrapper(publicKey string, secretKey string) ExchangeWrapper {
```

Comment:
This exchange has no support for deposit addresses, please add them as param in this function to allow depositAddresses map to be set
``` go
func NewKucoinWrapper(publicKey string, secretKey string, depositAddresses map[string]string) ExchangeWrapper {
    // ...
    depositAddresses : depositAddresses,
}
```